web-ui/src/components/feedback_submit_form/FeedbackSubmitForm.jsx

```diff
@@ -78,63 +73,149 @@ const propTypes = {
   requestId: PropTypes.string.isRequired,
 }
 
-const FeedbackSubmitForm = (props) => {
+const FeedbackSubmitForm = ({ requesteeName, requestId }) => {
+  const { state, dispatch } = useContext(AppContext);
+  const csrf = selectCsrfToken(state);
   const classes = useStyles();
-  const handleClick = () => history.push(`/feedback/submit/confirmation/?request=${props.requestId}`);
   const [isReviewing, setIsReviewing] = useState(false);
   const history = useHistory();
+  const [questionAnswerPairs, setQuestionAnswerPairs] = useState([])
+  let currentlyBeingEdited = -1
+
+
+  const updateAnswer = debounce(async () => {
+    if (csrf) {
+      const res = await updateSingleAnswer(questionAnswerPairs[currentlyBeingEdited].answer, csrf)
+      return res;
+    }
+  }, 2000)
+
+  async function updateAllAnswers(){
+    let answers= [];
+    for (let i = 0; i < questionAnswerPairs.length; ++i) {
+      answers.push(questionAnswerPairs[i].answer)
+    }
+    const res = await saveAllAnswers(answers, csrf)
+    return res;
+  }
+
+  const onSubmitHandler =() => {
+    updateAllAnswers().then((res) => {
+      for (let i = 0; i < res.length; ++i ) {
+        if (res[i].error) {
+          dispatch({
+            type: UPDATE_TOAST,
+            payload: {
+              severity: "error",
+              toast: res[i].error,
+            },
+          });
+          return;
+        }
+      }
+      history.push(`/feedback/submit/confirmation/?request=${requestId}`)
+    })
+  }
```

Comment:
We'll also need to update the feedback request when the form is submitted:
- `status` changes to `"submitted"`
- `submitDate` changes to the current date

Otherwise the user can keep going back to the form even after they have submitted it. There are currently checks in `FeedbackSubmitPage.jsx` that only render the form if the feedback request hasn't been submitted yet.

We can split this into a separate issue if you'd prefer

Reply:
Fixed to update request upon successful answer submit